pkg/services/ngalert/api/api_alertmanager.go

```diff
@@ -504,19 +362,26 @@ func statusForTestReceivers(v []notifier.TestReceiverResult) int {
 }
 
 func (srv AlertmanagerSrv) AlertmanagerFor(orgID int64) (Alertmanager, *response.NormalResponse) {
-	am, err := srv.mam.AlertmanagerFor(orgID)
+	am, err := srv.configs.AlertmanagerFor(orgID)
 	if err == nil {
 		return am, nil
 	}
 
-	if errors.Is(err, notifier.ErrNoAlertmanagerForOrg) {
-		return nil, response.Error(http.StatusNotFound, err.Error(), err)
-	}
-
-	if errors.Is(err, notifier.ErrAlertmanagerNotReady) {
-		return am, response.Error(http.StatusConflict, err.Error(), err)
+	if resp := tryMapAMLookupError(err); resp != nil {
```

Comment:
This is the only place in the PR where we really adjusted logic rather than just splits/moves. This is a POI for reviewers.

Reply:
Agree, this was definitely a rough edge. It didn't gain much.